### PR TITLE
Add data source health check

### LINKED ISSUE #62: Getting All datasources health with response code 200
How we can run health checks using grafana http api. Is there any way we can get health check information for all data sources while we're retrieving data sources.

### PR DESCRIPTION
**Pull request by asimonok**
Resolves #62 

Add query parameter to check ds health. Then data sources with status 200 can be filtered in transformations

### CODE REVIEW

File: src/api/datasources.ts [hunk 1]
```diff
@@ -36,79 +36,118 @@ export class DataSources extends BaseApi {
...
+
     if (!datasources.length) {
       return [];
     }
 
+    const datasourcesHealth: number[] = [];
+
+    if (query.datasourceHealth) {
+      for (const datasource of datasources) {
+        datasourcesHealth.push(await this.getHealthStatus(datasource.uid));
```
Comment (mikhail-vl) [inline src/api/datasources.ts:70]: Should it be Promise.all to speed up if there are many of them?

File: src/api/datasources.ts [hunk 2]
```diff
@@ -36,79 +36,118 @@ export class DataSources extends BaseApi {
...
+   * Get Health Status
+   */
+  private getHealthStatus = async (id: string): Promise<number> => {
+    /**
+     * Fetch
+     */
+    const response = await lastValueFrom(
+      getBackendSrv().fetch<DataSourceHealthMessage>({
+        method: 'GET',
+        url: `${this.api.instanceSettings.url}/api/datasources/uid/${id}/health`,
```
Comment (mikhail-vl) [inline src/api/datasources.ts:49]: Business Satellite returns `{"message":"Plugin unavailable","messageId":"plugin.unavailable","statusCode":500,"traceID":""}` when checking it's health.

**Review by mikhail-vl — CHANGES_REQUESTED** [review]
Business Satellite returns error for it's health-check:

Screenshot 2024-07-12 at 3 14 52 PM

File: src/api/datasources.ts (hunk 1 above)
Comment (asimonok) [reply src/api/datasources.ts]: fixed

File: src/api/datasources.ts (hunk 2 above)
Comment (asimonok) [reply src/api/datasources.ts:49]: fixed

**Review by mikhail-vl — APPROVED** [review]
Looks great!

**Pull request merged**